From ca89c49fa2c5cbb3945897046f33eed9f7da846c Mon Sep 17 00:00:00 2001
From: Jiang Jiang <jiang.jiang@nokia.com>
Date: Tue, 3 Jul 2012 10:17:49 +0200
Subject: [PATCH] Fix cursor truncate to include line position

Since we could have moved the line position (QTextLine::setPosition),
the truncating position should be adjusted with that.

Change-Id: Ie1acd4a1b6a4adfbeeb9ce8ee2dfa19d992470e8
Reviewed-by: Eskil Abrahamsen Blomfeldt <eskil.abrahamsen-blomfeldt@nokia.com>
---
 src/gui/text/qtextlayout.cpp |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gui/text/qtextlayout.cpp b/src/gui/text/qtextlayout.cpp
index 16f7150..52f2793 100644
--- ./src/gui/text/qtextlayout.cpp
+++ ./src/gui/text/qtextlayout.cpp
@@ -2616,8 +2616,8 @@ qreal QTextLine::cursorToX(int *cursorPos, Edge edge) const
         x += eng->offsetInLigature(si, pos, end, glyph_pos);
     }
 
-    if (eng->option.wrapMode() != QTextOption::NoWrap && x > line.width)
-        x = line.width;
+    if (eng->option.wrapMode() != QTextOption::NoWrap && x > line.x + line.width)
+        x = line.x + line.width;
 
     *cursorPos = pos + si->position;
     return x.toReal();
-- 
1.7.10

